integration/common/viewkey/viewkey.go

```diff
@@ -0,0 +1,61 @@
+package viewkey
+
+import (
+	"crypto/ecdsa"
+	"encoding/hex"
+
+	"github.com/ethereum/go-ethereum/accounts"
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/ethereum/go-ethereum/crypto/ecies"
+	"github.com/obscuronet/go-obscuro/go/enclave/rpcencryptionmanager"
+	"github.com/obscuronet/go-obscuro/go/rpcclientlib"
+	"github.com/obscuronet/go-obscuro/go/wallet"
+)
+
+// GenerateAndRegisterViewingKey take an obscuro client and a wallet, it generates a keypair, simulates signing the key,
+//	configures it on the client and registers the viewing key with the enclave
```

Comment:
// GenerateAndRegisterViewingKey takes an obscuro client and a wallet, generates a keypair, simulates signing the key,
//	configures the (which key?) key on the client and registers the viewing key with the enclave

Can this be tweaked a nudge ?

Reply:
Yeah that was confusing, have added a bunch more context, might be overkill but better to be too helpful than too vague I guess haha.